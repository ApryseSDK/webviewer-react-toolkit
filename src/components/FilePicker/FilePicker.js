'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.FilePicker = void 0;
var tslib_1 = require('tslib');
var classnames_1 = tslib_1.__importDefault(require('classnames'));
var react_1 = tslib_1.__importDefault(require('react'));
var EditableText_1 = require('../EditableText');
var IconButton_1 = require('../IconButton');
var Icon_1 = require('../Icon');
var FilePicker = function(_a) {
  var items = _a.items,
    className = _a.className,
    props = tslib_1.__rest(_a, ['items', 'className']);
  var filePickerClass = (0, classnames_1.default)(
    'ui__base ui__filePicker',
    className,
  );
  return react_1.default.createElement(
    'div',
    tslib_1.__assign({}, props, { className: filePickerClass }),
    items.map(function(item) {
      return react_1.default.createElement(
        'div',
        {
          className: (0, classnames_1.default)(
            'ui__filePicker__file',
            item.className,
          ),
          key: item.key,
        },
        react_1.default.createElement(EditableText_1.EditableText, {
          className: 'ui__filePicker__file__text',
          value: item.name,
          onSave: item.onRename,
          locked: !item.onRename,
        }),
        item.onDelete
          ? react_1.default.createElement(
              IconButton_1.IconButton,
              {
                className: 'ui__filePicker__file__delete',
                onClick: item.onDelete,
              },
              react_1.default.createElement(Icon_1.Icon, { icon: 'Close' }),
            )
          : undefined,
      );
    }),
  );
};
exports.FilePicker = FilePicker;
//# sourceMappingURL=FilePicker.js.map
