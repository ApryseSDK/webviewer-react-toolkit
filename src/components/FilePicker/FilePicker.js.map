{"version":3,"file":"FilePicker.js","sourceRoot":"","sources":["FilePicker.tsx"],"names":[],"mappings":";;;;AAAA,kEAAoC;AACpC,wDAA6D;AAC7D,gDAAkE;AAClE,4CAA2C;AAC3C,gCAA+B;AAcxB,IAAM,UAAU,GAAwB,UAAC,EAA8B;IAA5B,IAAA,KAAK,WAAA,EAAE,SAAS,eAAA,EAAK,KAAK,sBAA5B,sBAA8B,CAAF;IAC1E,IAAM,eAAe,GAAG,IAAA,oBAAU,EAAC,yBAAyB,EAAE,SAAS,CAAC,CAAC;IAEzE,OAAO,CACL,0DAAS,KAAK,IAAE,SAAS,EAAE,eAAe,KACvC,KAAK,CAAC,GAAG,CAAC,UAAC,IAAI,IAAK,OAAA,CACnB,uCAAK,SAAS,EAAE,IAAA,oBAAU,EAAC,sBAAsB,EAAE,IAAI,CAAC,SAAS,CAAC,EAAE,GAAG,EAAE,IAAI,CAAC,GAAG;QAC/E,8BAAC,2BAAY,IACX,SAAS,EAAC,4BAA4B,EACtC,KAAK,EAAE,IAAI,CAAC,IAAI,EAChB,MAAM,EAAE,IAAI,CAAC,QAAQ,EACrB,MAAM,EAAE,CAAC,IAAI,CAAC,QAAQ,GACtB;QACD,IAAI,CAAC,QAAQ,CAAC,CAAC,CAAC,CACf,8BAAC,uBAAU,IAAC,SAAS,EAAC,8BAA8B,EAAC,OAAO,EAAE,IAAI,CAAC,QAAQ;YACzE,8BAAC,WAAI,IAAC,IAAI,EAAC,OAAO,GAAG,CACV,CACd,CAAC,CAAC,CAAC,SAAS,CACT,CACP,EAdoB,CAcpB,CAAC,CACE,CACP,CAAC;AACJ,CAAC,CAAC;AAtBW,QAAA,UAAU,cAsBrB"}