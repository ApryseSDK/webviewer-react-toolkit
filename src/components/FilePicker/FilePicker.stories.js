'use strict';
Object.defineProperty(exports, '__esModule', { value: true });
exports.Basic = void 0;
var tslib_1 = require('tslib');
var addon_knobs_1 = require('@storybook/addon-knobs');
var react_1 = tslib_1.__importDefault(require('react'));
var action_1 = require('../../storybook-helpers/action/action');
var FilePicker_1 = require('../FilePicker');
var README_md_1 = tslib_1.__importDefault(require('./README.md'));
exports.default = {
  title: 'Components/FilePicker',
  component: FilePicker_1.FilePicker,
  parameters: { readme: README_md_1.default },
};
var Basic = function() {
  var onRename = (0, addon_knobs_1.boolean)('has onRename', false);
  var onDelete = (0, addon_knobs_1.boolean)('has onDelete', false);
  var long = (0, addon_knobs_1.boolean)('has long name', false);
  return react_1.default.createElement(
    'div',
    { style: { width: 200 } },
    react_1.default.createElement(FilePicker_1.FilePicker, {
      items: [
        {
          key: 0,
          name: 'file_1'.concat(
            long ? '_with_long_name_to_show_cut_off' : '',
            '.pdf',
          ),
          onRename: onRename
            ? (0, action_1.action)('onRename file 1')
            : undefined,
          onDelete: onDelete
            ? (0, action_1.action)('onDelete file 1')
            : undefined,
        },
        {
          key: 1,
          name: 'file_2'.concat(
            long ? '_with_long_name_to_show_cut_off' : '',
            '.pdf',
          ),
          onRename: onRename
            ? (0, action_1.action)('onRename file 2')
            : undefined,
          onDelete: onDelete
            ? (0, action_1.action)('onDelete file 2')
            : undefined,
        },
      ],
    }),
  );
};
exports.Basic = Basic;
//# sourceMappingURL=FilePicker.stories.js.map
