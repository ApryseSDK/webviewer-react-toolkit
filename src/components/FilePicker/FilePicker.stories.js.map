{"version":3,"file":"FilePicker.stories.js","sourceRoot":"","sources":["FilePicker.stories.tsx"],"names":[],"mappings":";;;;AAAA,sDAAiD;AACjD,wDAA0B;AAC1B,gEAA+D;AAC/D,4CAA2C;AAC3C,kEAAiC;AAEjC,kBAAe,EAAE,KAAK,EAAE,uBAAuB,EAAE,SAAS,EAAE,uBAAU,EAAE,UAAU,EAAE,EAAE,MAAM,qBAAA,EAAE,EAAE,CAAC;AAE1F,IAAM,KAAK,GAAG;IACnB,IAAM,QAAQ,GAAG,IAAA,qBAAO,EAAC,cAAc,EAAE,KAAK,CAAC,CAAC;IAChD,IAAM,QAAQ,GAAG,IAAA,qBAAO,EAAC,cAAc,EAAE,KAAK,CAAC,CAAC;IAChD,IAAM,IAAI,GAAG,IAAA,qBAAO,EAAC,eAAe,EAAE,KAAK,CAAC,CAAC;IAE7C,OAAO,CACL,uCAAK,KAAK,EAAE,EAAE,KAAK,EAAE,GAAG,EAAE;QACxB,8BAAC,uBAAU,IACT,KAAK,EAAE;gBACL;oBACE,GAAG,EAAE,CAAC;oBACN,IAAI,EAAE,gBAAS,IAAI,CAAC,CAAC,CAAC,iCAAiC,CAAC,CAAC,CAAC,EAAE,SAAM;oBAClE,QAAQ,EAAE,QAAQ,CAAC,CAAC,CAAC,IAAA,eAAM,EAAC,iBAAiB,CAAC,CAAC,CAAC,CAAC,SAAS;oBAC1D,QAAQ,EAAE,QAAQ,CAAC,CAAC,CAAC,IAAA,eAAM,EAAC,iBAAiB,CAAC,CAAC,CAAC,CAAC,SAAS;iBAC3D;gBACD;oBACE,GAAG,EAAE,CAAC;oBACN,IAAI,EAAE,gBAAS,IAAI,CAAC,CAAC,CAAC,iCAAiC,CAAC,CAAC,CAAC,EAAE,SAAM;oBAClE,QAAQ,EAAE,QAAQ,CAAC,CAAC,CAAC,IAAA,eAAM,EAAC,iBAAiB,CAAC,CAAC,CAAC,CAAC,SAAS;oBAC1D,QAAQ,EAAE,QAAQ,CAAC,CAAC,CAAC,IAAA,eAAM,EAAC,iBAAiB,CAAC,CAAC,CAAC,CAAC,SAAS;iBAC3D;aACF,GACD,CACE,CACP,CAAC;AACJ,CAAC,CAAC;AAzBW,QAAA,KAAK,SAyBhB"}