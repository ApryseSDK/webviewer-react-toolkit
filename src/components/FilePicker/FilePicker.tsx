import classnames from 'classnames';
import React, { FC, HTMLAttributes, ReactText } from 'react';
import { EditableTextProps, EditableText } from '../EditableText';
import { IconButton } from '../IconButton';
import { Icon } from '../Icon';

export interface FilePickerItem {
  key: ReactText;
  name: string;
  onRename?: EditableTextProps['onSave'];
  onDelete?: () => void;
  className?: string;
}

export interface FilePickerProps extends HTMLAttributes<HTMLDivElement> {
  items: FilePickerItem[];
}

export const FilePicker: FC<FilePickerProps> = ({ items, className, ...props }) => {
  const filePickerClass = classnames('ui__base ui__filePicker', className);

  return (
    <div {...props} className={filePickerClass}>
      {items.map(item => (
        <div className={classnames('ui__filePicker__file', item.className)} key={item.key}>
          <EditableText
            className="ui__filePicker__file__text"
            value={item.name}
            onSave={item.onRename}
            locked={!item.onRename}
          />
          {item.onDelete ? (
            <IconButton className="ui__filePicker__file__delete" onClick={item.onDelete}>
              <Icon icon="Close" />
            </IconButton>
          ) : (
            undefined
          )}
        </div>
      ))}
    </div>
  );
};
