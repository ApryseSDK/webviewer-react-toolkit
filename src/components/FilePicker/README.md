Organizes multiple files in a column, with options to rename or delete.
