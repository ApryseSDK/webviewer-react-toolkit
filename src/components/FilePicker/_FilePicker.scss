.ui__filePicker {
  &__file {
    display: flex;
    align-items: center;
    min-height: 28px;

    &__text {
      flex: 1 1 auto;
    }

    &__delete.ui__button.ui__iconButton {
      height: 28px;
      width: 28px;
      min-height: 28px;
      min-width: 28px;
      margin-left: $padding-tiny;
    }
  }
}
