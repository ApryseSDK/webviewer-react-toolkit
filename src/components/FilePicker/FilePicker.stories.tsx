import React from 'react';
import { action } from '../../storybook-helpers/action/action';
import { FilePicker } from '../FilePicker';
import { FilePickerItem } from './FilePicker';
import readme from './README.md';

export default { title: 'Components/FilePicker', component: FilePicker, parameters: { readme } };

const getFiles = (editable = false): FilePickerItem[] => [
  {
    key: 0,
    name: 'file_1.pdf',
    onRename: editable ? action('onRename file 1') : undefined,
    onDelete: editable ? action('onDelete file 1') : undefined,
  },
  {
    key: 1,
    name: 'file_2.pdf',
    onRename: editable ? action('onRename file 2') : undefined,
    onDelete: editable ? action('onDelete file 2') : undefined,
  },
];

export const Basic = () => (
  <div style={{ width: 200 }}>
    <FilePicker items={getFiles()} />
  </div>
);

export const Editable = () => (
  <div style={{ width: 200 }}>
    <FilePicker items={getFiles(true)} />
  </div>
);
