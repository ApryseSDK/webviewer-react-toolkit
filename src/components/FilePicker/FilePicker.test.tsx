import { shallow } from 'enzyme';
import React from 'react';
import { FilePicker } from '../FilePicker';

describe('FilePicker component', () => {
  it('renders its contents', () => {
    const filePicker = shallow(<FilePicker items={[]} />);
    expect(filePicker.find('.ui__filePicker')).toHaveLength(1);
  });

  it('snapshot renders default filePicker', () => {
    const filePicker = shallow(<FilePicker items={[]} />);
    expect(filePicker).toMatchSnapshot();
  });
});
