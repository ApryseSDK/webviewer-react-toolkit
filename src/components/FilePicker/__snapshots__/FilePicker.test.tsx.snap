// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilePicker component snapshot renders default filePicker 1`] = `
<div
  className="ui__base ui__filePicker"
/>
`;
